Til-Chapter-3 (#152)
scionassociation/scion-cppki_I-D: draft-dekater-scion-pki.md

Reviewer: What is missing here is a recommendation on:
- How many authorative ASes should have?
- How do we chose them (is it okay if they are voting ASes?)
- Do authorative ASes require additional infrastructure?

And some explanation what their purpose is:
- Why do we need them? They are not part of the trust model... 
- If there is no purpose beyond just giving the concept a name, maybe state just that. As it is, the reader is left wondering whether there is importance or not.

Reviewer: Maybe from the AS perspective:
- What are the advantages/disadvantages for an AS of being authorative?

Reviewer 2: > Why do we need them? They are not part of the trust model...
 
 This is already explained in the terminology
 >    *Authoritative AS*: Authoritative ASes are those ASes in an ISD that
   always have the latest TRCs of the ISD.  As a consequence,
   authoritative ASes also start the announcement of a TRC update. 
 
 And also in this paragraph you quote:
 
 > Authoritative ASes are those ASes in an ISD that always possess the latest TRCs for the ISD and initiate TRC update announcements.
 
For the other points, I am not sure if this fits best 
3.2.9.  authoritativeASes
4.1.2.  TRC Update Discovery (we should mention the role of authoritative ASes here!)
5.  Deployment Considerations (maybe we could say something about the availability requirements)

Since this affects 3 sections tracking this in https://github.com/scionassociation/scion-cppki_I-D/issues/155

Reviewer: This section (Section 3) explains concepts such as "trust updates" and "the signing ceremony", but it does not describe "trust reset", which is a very related concept and basically the middle ground..?.
I think this is missing.
Section 1.4.2 gives an overview over trust reset. Section 6.1.1 goes into details.

Reviewer: This comment can be resolved, I moved it to the suggestion above.

Reviewer: ```suggestion
## Trust Reset

If the ISD's TRC has been compromised, it is necessary for an ISD to re-establish the trust root. This is possible with a process called trust reset (if permitted by the ISD's trust policy). In this case, a new base TRC is created.

If a TRC is compromised, it means that the root keys or voting keys are compromised. If the number of compromised keys is smaller than the voting quorum, the TRC can be updated as described in [Section 3.5](https://scionassociation.github.io/scion-cppki_I-D/draft-dekater-scion-pki.html#update). 

A trust reset is only required in the case the number of compromised keys at the same time is greater or equal than the TRC's quorum (see [Section 3.2.7](https://scionassociation.github.io/scion-cppki_I-D/draft-dekater-scion-pki.html#quorum)), and a invalid update has been produced and distributed in the network.

```
This section (Section 3) explains concepts such as "trust updates" and "the signing ceremony", but it does not describe "trust reset", which is a very related concept and basically the middle ground..?.
I think this is missing.
Section 1.4.2 gives an overview over trust reset. Section 6.1.1 goes into details.

The text above is an amalgamation of 1.4.1 and 6.1.1. Section links need to be fixed.

Reviewer 2: Postponing to separate issue since I have to touch 3 sections to have proper cross references
https://github.com/scionassociation/scion-cppki_I-D/issues/158 
I copied your proposal over

- Valid: The validity period of a TRC is defined in the TRC itself, in the `validity` field (see [](#validity-trc)). A TRC is considered valid if the current time falls within its validity period.
- Active: An active TRC is a valid TRC that can be used for verifying certificate signatures. This is either the latest TRC or the predecessor TRC if it is still in its grace period (as defined in the `gracePeriod` field of the new TRC, see [](#grace)). No more than two TRCs can be active at the same time for any ISD.

> - Invalid: The TRC is neither valid nor active.

Reviewer 2: ```suggestion
- Invalid: The TRC is considered invalid if the current time falls outside its validity period.
```

Id does not make sense to mention "active" since the active time window is a subset of the valid time window

An active TRC is a valid TRC that can be used for verifying certificate signatures. The time period during which a TRC is active can be shorter than the time period during which the TRC is valid. For more information, see [](#trc-states).

The `validity` field consists of a sequence of a `notBefore` and a `notAfter` date, both encoded as `GeneralizedTime`.

> All TRCs MUST have a well-defined expiration date. SCION implementations MUST NOT create TRCs that use GeneralizedTime` value "99991231235959Z", and verifiers MUST reject such a TRC.

Reviewer 3: That looks off

Reviewer 2: ```suggestion
All TRCs MUST have a well-defined expiration date. SCION implementations MUST NOT create TRCs that use `GeneralizedTime` value "99991231235959Z", and verifiers MUST reject such a TRC.
```

The `votingQuorum` field defines the number of necessary votes on a successor TRC to make it verifiable.

A voting quorum greater than one will prevent a single entity from creating a malicious TRC update.

> A voting quorum lower than the number of Voting ASes ensures that votes can be cast even if some of the voters are unavailable.

Reviewer 2: @tzaeschke like it ?

Reviewer: Yes, I like it.
